source/FormationPositioner.h

```diff
@@ -0,0 +1,103 @@
+/* FormationPositioner.h
+Copyright (c) 2019-2022 by Peter van der Meer
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef FORMATION_POSITIONER_H_
+#define FORMATION_POSITIONER_H_
+
+#include "Angle.h"
+
+#include <map>
+#include <memory>
+#include <vector>
+
+class Body;
+class FormationPattern;
+class Ship;
+
+
+
+// Represents an active formation for a set of spaceships. Assigns each ship
+// to a position (Point) in the formation.
+class FormationPositioner {
+public:
+	// Initializer based on the formation pattern to follow.
+	FormationPositioner(const Body *formationLead, const FormationPattern *pattern);
+
+	// Start/reset/initialize for a (new) round of formation position calculations
+	// for a formation around the ship given as parameter.
```

Comment:
I see no ship parameter here. Or did you mean something else?
```suggestion
	// Start/reset/initialize for a (new) round of formation position calculations.
```